commit bc6e8bfa1cd91ec3946b0bd57829ff9e6a733907 (refs/changes/85/1485/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-02-09T16:40:55-08:00 (2 years, 8 months ago)
    
    TOOLS-1689 sdcadm hang on exit: suspect urclient leaving handles open after .close() + cmdln softexit
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/cli/index.js b/lib/cli/index.js
index 2fa2f1d..e98fbe1 100644
--- a/lib/cli/index.js
+++ b/lib/cli/index.js
@@ -467,8 +467,10 @@ module.exports = CLI;
 
 if (require.main === module) {
     var cli = new CLI();
-    cmdln.main(cli, process.argv, {
+    cmdln.main(cli, {
+        argv: process.argv,
         showCode: true,
-        showErr: true
+        showErr: true,
+        finale: 'exit'
     });
 }
diff --git a/package.json b/package.json
index bfa6be5..1630a4b 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
     "assert-plus": "^1.0.0",
     "backoff": "2.5.0",
     "bunyan": "0.23.1",
-    "cmdln": "4.2.0",
+    "cmdln": "4.2.1",
     "extsprintf": "^1.3.0",
     "kthxbai": "~0.4.0",
     "joyent-schemas": "git+https://github.com/joyent/schemas.git#385e6eb",
